packages/klevu-ui/src/components/klevu-init/klevu-init.tsx

```diff
@@ -119,6 +124,46 @@ export class KlevuInit {
       const data = await KlevuKMCSettings()
       window["klevu_ui_kmc_settings"] = data.root
     }
+
+    if (this.settings?.renderPrice === undefined) {
+      this.settings = {
+        ...(this.settings || {}),
+        renderPrice: this.#renderPriceKMCSettings,
+      }
+    }
+    if (this.settings) {
+      window["klevu_ui_settings"] = this.settings
+    }
+  }
+
+  #renderPriceKMCSettings(amount: string | number, currency: string) {
+    // Use KMC settings to format price
+    const kmcSettings = getKMCSettings()
+    const priceSettings: PriceFormatSettings | undefined = kmcSettings?.klevu_uc_userOptions.priceFormatter
+    if (priceSettings) {
```

Comment:
I think this needs to be inverted so that we first check the edge cases ([Read this for more info](https://softwareengineering.stackexchange.com/questions/220461/keep-indentation-level-low))

And maybe we should return undefined so that default price formatting is used instead of just plain amount.

```suggestion
    if (!priceSettings) {
      return undefined
    }
```

Reply:
since this is the return value of the function, returning undefined here wont help. I have refactored the logic to have the check while assigning the function to renderPrice, let me know if this works.